frontend/public/components/app.jsx

```diff
@@ -230,14 +230,17 @@ graphQLReady.onReady(() => {
   // Used by GUI tests to check for unhandled exceptions
   window.windowError = null;
   window.onerror = (message, source, lineno, colno, error) => {
+    const { stack } = error;
+    const formattedStack = stack?.replace(/\\n/g, '\n');
+    window.windowError = `window error detected: ${message} ${formattedStack || ''}`;
     // eslint-disable-next-line no-console
-    console.error('Uncaught error', error);
-    window.windowError = error;
+    console.error(window.windowError);
   };
   window.onunhandledrejection = (promiseRejectionEvent) => {
+    const { reason } = promiseRejectionEvent;
+    window.windowError = reason;
```

Comment:
Should this be something like

```suggestion
    window.windowError = `unhandled promise rejection: ${reason}`;
```

Reply:
fixed